splink/term_frequencies.py

```diff
@@ -75,14 +75,58 @@ def _join_tf_to_input_df_sql(linker: Linker):
     return sql
 
 
-def term_frequencies_from_concat_with_tf(input_column):
+def _join_new_table_to_df_concat_with_tf_sql(linker: Linker, new_tablename) -> str:
```

Comment:
Previously, we were reusing logic in `_join_tf_to_input_df_sql` by performing string replacement operations on its outputs.  

This was confusing.  Also these two functions have diverged more with this PR, so it's no longer easy to have a single function